schema/tx_schema.md

```diff
@@ -0,0 +1,155 @@
+## Overview
+
+Signal size can be automatically calculated or explicitly specified. Cannot (currently) specify signal position.
+
+## Default TX message definition fields
+
+```json
+{
+    "message_name": {
+        "signals": {
+            "signal_name": {
+                "resolution": 1,
+                "offset": 0,
+                "signed": false,
+                "min": 0
+            }
+        }
+    }
+}
+```
+
+## Absolute minimum message definitions
+
+Given the default fields there are two options of absolute minimum TX message definitions:
+```json
+{
+    "message_name": {
+        "message_id": 123,
+        "signals": {
+            "signal_name_1": {
+                "bits": 14
+            },
+            "signal_name_2": {
+                "max": 100
+            }
+        }
+    }
+}
+```
+
+## Associate an enum definition with a signal
+
+The required signal size/sign is automatically determined, even if the enum is negative. It's illegal to override this by defining `min`, `max` and/or `signed`.
+
+Note that this is legal:
+```json
+{
+    "message_name": {
+        "signals": {
+            "signal_name_1": {
+                "enum": "example_enum_1"
+            }
+        }
+    }
+}
+```
+
+Whereas this is illegal, even if it correctly describes the bounds of the enum:
+```json
+{
+    "message_name": {
+        "signals": {
+            "signal_name_2": {
+                "enum": "example_enum_2",
+                "min": -3,
+                "max": 3,
+                "signed": true
+            }
+        }
+    }
+}
+```
+
+## Signal size
+
+Define any valid combination of `min`, `max`, `bits`, and `offset`. To generate optimally sized signals, provide as little information as possible.
+
+```json
+{
+    "message_name": {
+        "signals": {
+            "signal_name_1": {
+                "min": 0,
+                "max": 10
+            },
+            "signal_name_2": {
+                "bits": 4
+            },
+            "signal_name_3": {
+                "min": -3,
+                "bits": 2
+            },
+            "signal_name_4": {
+                "min": -500,
+                "max": 1000,
+                "offset": 500,
+                "bits": 12

```

Comment:
is this offset currently manually defined?

Reply:
what do you mean by manually defined? if you scroll up you'll see the default offset is 0

Reply:
it is typically but this is an exception where the default offset should be halfway at `250` right... perhaps i should be clearer about this

- [ ] clarify hardcoded offset - create an offset section and talk about how offset defaults to halfway between min and max

Reply:
but there should be no reason to add this offset then right? shouldn't the user just specify a range of 0 and 1500?

Reply:
i could be a little confused tho so plz clarify if im msising sth

Reply:
removing offset field - ignore